docs/BreakingChanges/027 Data Definition Language (DDL) APIs.md

```diff
@@ -0,0 +1,28 @@
+## 27: Data Definition Language (DDL) APIs
```

Comment:
{
  "Id": "27",
  "Title": "Data Definition Language (DDL) APIs",
  "Details": "The behavior of DDL APIs when AttachDBFilename is specified has changed as follows:\n* Connection strings need not specify an Initial Catalog value. Previously, both AttatchDBFilename and Initial Catalog were required.\n* If both AttatchDBFilename and Initial Catalog are specified and the given MDF file exists, the ObjectContext.DatabaseExists method returns true. Previously, it returned false.\n* If both AttatchDBFilename and Initial Catalog are specified and the given MDF file exists, calling the ObjectContext.DeleteDatabase method deletes the files.\n* If ObjectContext.DeleteDatabase is called when the connection string specifies an AttachDBFilename value with an MDF that doesn't exist and an Initial Catalog that doesn't exist, the method throws an InvalidOperationException exception. Previously, it threw a SqlException exception.",
  "Suggestion": "These changes make it easier to build tools and applications that use the DDL APIs. These changes can affect application compatibility in the following scenarios:\n- The user writes code that executes a DROP DATABASE command directly instead of calling ObjectContext.DeleteDatabase if ObjectContext.DatabaseExists returns true. This breaks existing code If the database is not attached but the MDF file exists.\n- The user writes code that expects the ObjectContext.DeleteDatabase method to throw a SqlException exception rather than an InvalidOperationException exception when the Initial Catalog and MDF file don't exist.",
  "Link": "https://msdn.microsoft.com/en-us/library/hh367887(v=vs.110).aspx#sql",
  "Markdown": null,
  "ApplicableApis": [
    "Not detectable via API analysis"
  ],
  "Related": null,
  "VersionBroken": "4.5",
  "VersionFixed": null,
  "IsBuildTime": false,
  "IsQuirked": false,
  "IsSourceAnalyzerAvailable": false,
  "BugLink": null,
  "Notes": null,
  "IsRetargeting": false,
  "ImpactScope": 2
}

**Parsed APIs:**
- [ ] Not detectable via API analysis